src/test/java/CarRacingTest.java

```diff
@@ -1,22 +1,134 @@
 import org.junit.jupiter.params.provider.CsvSource;
+import record.CarRecord;
+import record.Records;
+import strategy.RandomStrategy;
+
+import java.util.Arrays;
+import java.util.List;
 
 import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.assertj.core.api.SoftAssertions.assertSoftly;
 
 class CarRacingTest {
 
-    @ParameterizedTest(name = "전진 조건이 {0}이면 차의 주행거리가 {1} 증가한다.")
+    private final int TRIAL = 5;
+    private final int TOTAL_RECORD = 10;
+
+    @ParameterizedTest(name = "전진 전략이 {0}이면 차의 주행거리가 {1} 증가한다.")
     @CsvSource(value = {"true,1", "false,0"})
     void 전진조건이_참이면_차가_전진한다(boolean condition, int result) {
 
         // given
-        Car car = new Car();
+        Car car = new Car("car");
 
         // when
-        car.go(condition);
+        car.go(() -> condition);
 
         // then
         assertThat(car.distance()).isEqualTo(result);
     }
 
+    @Test
+    @DisplayName("차의 이름은 5자 초과가 될 수 없다.")
+    void 차의_이름이_5자_초과이면_예외가_발생한다() {
+
+        // given
+        String testName = "abcdef";
+
+        // then
+        assertThatThrownBy(() -> new Car(testName))
+                .isInstanceOf(IllegalArgumentException.class)
+                .hasMessage("이름이 5자 이상이 될 수 없습니다! : " + testName);
+    }
+
+    @Test
+    void 주행_거리가_긴_자동차가_우승한다() {
+
+        // given
+        Car loser = new Car("alex");
+        Car winner = new Car("jack");
+        loser.go(() -> false);
+        winner.go(() -> true);
+
+        List<Car> carList = Arrays.asList(loser, winner);
+        Cars cars = new Cars(carList, new RandomStrategy());
+        Records records = new Records();
+
+        // when
+        records.add(cars);
+        records.addWinners();
+
+        List<CarRecord> result = records.winners();
+
+        // then
+        assertSoftly(softAssertions -> {
+            softAssertions.assertThat(result).hasSize(1);
+            softAssertions.assertThat(result.get(0).name()).isEqualTo("jack");
+            softAssertions.assertThat(result.get(0).distance())
+                    .isGreaterThan(loser.distance());
+        });
+    }
+
+    @Test
+    void 우승자는_한명_이상일_수_있다() {
+
+        // given
+        Car firstWinner = new Car("alex");
+        Car secondWinner = new Car("jack");
+        firstWinner.go(() -> true);
+        secondWinner.go(() -> true);
+
+        List<Car> carList = Arrays.asList(firstWinner, secondWinner);
+        Cars cars = new Cars(carList, new RandomStrategy());
+        Records records = new Records();
+
+        // when
+        records.add(cars);
+        records.addWinners();
+
+        List<CarRecord> result = records.winners();
+
+        // then
+        assertSoftly(softAssertions -> {
+            softAssertions.assertThat(result).hasSize(2);
+            softAssertions.assertThat(result.get(0).name()).isEqualTo("alex");
+            softAssertions.assertThat(result.get(1).name()).isEqualTo("jack");
```

Comment:
```suggestion
            softAssertions.assertThat(result).contains(new CarRecord("alex", 1));
            softAssertions.assertThat(result).contains(new CarRecord("jack", 1));
```

CarRecord equals를 재정의 했으니 객체를 비교하는 것이 어떨까요?